azure/durable_functions/models/TaskOrchestrationExecutor.py

```diff
@@ -0,0 +1,280 @@
+        self.initialize()
+
+    def initialize(self):
+        """Initialize the TaskOrchestrationExecutor for a new orchestration invocation."""
+        # The first task is just a placeholder to kickstart the generator.
+        # So it's value is `None`.
+        # TODO: need to set the `is_replaying` flag in here!
+        self.current_task: TaskBase = AtomicTask(-1, [])
+        self.current_task.set_value(is_error=False, value=None)
+
+        self.output: Any = None
+        self.exception: Optional[Exception] = None
+        self.orchestrator_returned: bool = False
+
+    def execute(self, context: DurableOrchestrationContext,
+                history: List[HistoryEvent], fn) -> str:
+        """Execute an orchestration using the orchestration history to evaluate Tasks and replay events.
+
+        Parameters
+        ----------
+        context : DurableOrchestrationContext
+            The user's orchestration context, to interact with the user code.
+        history : List[HistoryEvent]
+            The orchestration history, to evaluate tasks and replay events.
+        fn : function
+            The user's orchestration function.
+
+        Returns
+        -------
+        str
+            A JSON-formatted string of the user's orchestration state, payload for the extension.
+        """
+        self.context = context
+        evaluated_user_code = fn(context)
+
+        # If user code is a generator, then it uses `yield` statements (the DF API)
+        # and so we iterate through the DF history, generating tasks and populating
+        # them with values when the history provides them
+        if isinstance(evaluated_user_code, GeneratorType):
+            self.generator = evaluated_user_code
+            for event in history:
+                self.process_event(event)
+                if self.has_execution_completed:
+                    break
+
+        # Due to backwards compatibility reasons, it's possible
+        # for the `continue_as_new` API to be called without `yield` statements.
+        # Therefore, we explicitely check if `continue_as_new` was used before
+        # flatting the orchestration as returned/completed
+        elif not self.context.will_continue_as_new:
+            self.orchestrator_returned = True
+            self.output = evaluated_user_code
+        return self.get_orchestrator_state_str()
+
+    def process_event(self, event: HistoryEvent):
+        """Evaluate a history event.
+
+        This might result in updating some orchestration internal state deterministically,
+        to evaluating some Task, or have no side-effects.
+
+        Parameters
+        ----------
+        event : HistoryEvent
+            The history event to process
+        """
+        event_type = event.event_type
+        if event_type == HistoryEventType.ORCHESTRATOR_STARTED:
+            # update orchestration's deterministic timestamp
+            timestamp = event.timestamp
```

Comment:
I believe this is also where you would want to set is_replayed.

Reply:
Hmm, I'll look into this.
I believe the very first value of `is_replayed` is passed in as part of the durable-extension payload, and set here: https://github.com/Azure/azure-functions-durable-python/blob/1104a2080a0c99418e550fb11ca85f8733562ea4/azure/durable_functions/models/DurableOrchestrationContext.py#L32-L37

Assuming that were correct, we wouldn't need to set `is_replaying` in the line you suggest. But I'll triple check